### PR TITLE
added finalized block

### PR DESCRIPTION
**Pull request by Wayonb**

### CODE REVIEW

File: schemas/finalization/FinalizedBlock.cats [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# binary layout for finalization round
+struct FinalizationRound
+	# finalization epoch
+	epoch = FinalizationEpoch
+
+	# finalization point
+	point = FinalizationPoint
+
+# binary layout for finalization block header
+struct FinalizedBlock
```
Comment (Jaguar0625) [inline schemas/finalization/FinalizedBlock.cats:12]: i assume this is supposed to match what is sent over ZMQ?

File: schemas/finalization/FinalizedBlock.cats [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+# binary layout for finalization round
+struct FinalizationRound
+	# finalization epoch
+	epoch = FinalizationEpoch
+
+	# finalization point
+	point = FinalizationPoint
+
+# binary layout for finalization block header
```
Comment (Jaguar0625) [inline schemas/finalization/FinalizedBlock.cats:11]: finalization => finalized?

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]
please add to generate_all.sh

File: schemas/finalization/FinalizedBlock.cats (hunk 1 above)
Comment (Wayonb) [reply schemas/finalization/FinalizedBlock.cats:12]: Yes.  I can name it the same if you want (PackedFinalizedBlockHeader)?  maybe that will be clearer.

File: schemas/finalization/packed_finalized_block_header.cats [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# binary layout for packed finalized block header
+struct PackedFinalizedBlockHeader
+	# finalization round
+	round = FinalizationRound
+
+	# block height
+	height = Height
+
+	# proof hash
+	hash = Hash256
```
Comment (Jaguar0625) [inline schemas/finalization/packed_finalized_block_header.cats:13]: please add newline

File: schemas/types.cats [hunk 4]
```diff
@@ -2,6 +2,7 @@ using Amount = uint64
 using BlockDuration = uint64
 using BlockFeeMultiplier = uint32
 using Difficulty = uint64
+using FinalizationPoint = uint32
 using FinalizationEpoch = uint32
```
Comment (Jaguar0625) [inline schemas/types.cats:6]: please alphabetize

File: schemas/finalization/packed_finalized_block_header.cats [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import "types.cats"
```
Comment (Jaguar0625) [inline schemas/finalization/packed_finalized_block_header.cats:1]: unneeded ?

File: schemas/finalization/FinalizedBlock.cats (hunk 1 above)
Comment (Jaguar0625) [reply schemas/finalization/FinalizedBlock.cats]: i would probably name it `FinalizedBlockHeader` (all catbuffer types are implicitly packed).

File: scripts/schema_lists.sh [hunk 6]
```diff
@@ -3,6 +3,10 @@
 block_inputs=(
 	"block")
 
+finalization_inputs=(
+	"finalization/finalization_round.cats",
+	"finalization/finalized_block_header.cats")
```
Comment (Jaguar0625) [inline scripts/schema_lists.sh:8]: 1. no comma 
2. remove ".cats" file 

[this is causing travis to fail]

File: schemas/finalization/finalized_block_header.cats [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import "finalization/finalization_round.cats"
+
+# binary layout for packed finalized block header
```
Comment (Jaguar0625) [inline schemas/finalization/finalized_block_header.cats:3]: ~~packed~~

File: schemas/finalization/finalized_block_header.cats [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+struct FinalizedBlockHeader
+	# finalization round
+	round = FinalizationRound
+
+	# block height
```
Comment (Jaguar0625) [inline schemas/finalization/finalized_block_header.cats:8]: block => proof ?

**Review by Jaguar0625 — CHANGES_REQUESTED** [review]

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**